util/keyfile/src/bin/sample_keys_main.rs

```diff
@@ -33,9 +36,10 @@ fn main() {
     if config.general.fog_report_url.is_some() && spki.is_none() {
         panic!("Fog report url was passed, so fog is enabled, but no fog authority spki was provided. This is needed for the fog authority signature scheme. Use --fog-authority-root to pass a .pem file or --fog-authority-spki to pass base64 encoded bytes specifying this.")
     }
-
-    if config.general.fog_report_url.is_some() && config.general.fog_report_id.is_none() {
-        panic!("Fog report url was passed, so fog is enabled, but no fog report id was provided. This is needed for the fog authority signature scheme. Use --fog_report_id to pass an id string specifying this.")
+    let mut fog_report_id = config.general.fog_report_id.as_deref();
+    if config.general.fog_report_url.is_some() && fog_report_id.is_none() {
+        log::info!(logger, "Fog report url was passed, so fog is enabled, but no fog report id was provided. Using default value instead. Use --fog-report-id to pass string specifying this.");
+        fog_report_id = Some("");
```

Comment:
This can just be `default=""` in the clap markup, right?

Reply:
That seems correct yes. Will do.

Reply:
I think that means it should just be `String` and not `Option<String>`

Reply:
That would mean that there is no way to show the `info!` log, since you can't tell if passed explicitly or defaulted, but I don't think that matters.